### PR TITLE
OSDOCS-9846: adds 4.14.21 relnote MicroShift

### PR DESCRIPTION
**Pull request by ShaunaDiaz**
Version(s):
4.14 only

Issue:
[OSDOCS-9846](https://issues.redhat.com/browse/OSDOCS-9846)

Link to docs preview:
[4.14.21 release note](https://74574--ocpdocs-pr.netlify.app/microshift/latest/microshift_release_notes/microshift-4-14-release-notes.html#microshift-4-14-21-dp)

QE review:
- [x] N/A stock text, no bug text or new technical information

### CODE REVIEW

File: microshift_release_notes/microshift-4-14-release-notes.adoc [hunk 1]
```diff
@@ -485,4 +485,13 @@ Issued: 2024-04-08
 
 {product-title} release 4.14.20 is now available. The list of bug fixes that are included in the update is documented in the link:https://access.redhat.com/errata/RHBA-2024:1682[RHBA-2024:1682] advisory. The images that are included in the update are provided by the link:https://access.redhat.com/errata/RHSA-2024:1681[RHSA-2024:1681] advisory.
 
+For the latest images included with {microshift-short}, view the contents of the `microshift-release-info` RPM. See xref:../microshift_install/microshift-embed-in-rpm-ostree-offline-use.adoc#microshift-embed-microshift-image-offline-deployment_microshift-embed-rpm-ostree-offline-use[Embedding {microshift-short} containers for offline deployments].
+
+[id="microshift-4-14-21-dp"]
+=== RHBA-2024:1767 - {microshift-short} 4.14.21 bug fix and enhancement advisory
+
+Issued: 2024-04-18
```
Comment (lpettyjo) [inline microshift_release_notes/microshift-4-14-release-notes.adoc:493]: OK that date is in the future?

File: microshift_release_notes/microshift-4-14-release-notes.adoc [hunk 2]
```diff
@@ -485,4 +485,13 @@ Issued: 2024-04-08
...
+
+{product-title} release 4.14.21 is now available. The list of enhancements and bug fixes that are included in the update is documented in the link:https://access.redhat.com/errata/RHBA-2024:1767[RHBA-2024:1767] advisory. The images that are included in the update are provided by the link:https://access.redhat.com/errata/RHSA-2024:1765[RHSA-2024:1765] advisory.
```
Comment (lpettyjo) [inline microshift_release_notes/microshift-4-14-release-notes.adoc:495]: Both advisory links yield 404 errors. Will these links resovle correctly on the projected date?

**Review by lpettyjo — COMMENTED** [review]
Two comments. Otherwise, LGTM.

File: microshift_release_notes/microshift-4-14-release-notes.adoc (hunk 1 above)
Comment (ShaunaDiaz) [reply microshift_release_notes/microshift-4-14-release-notes.adoc:493]: Yes, that's the ART date for the upcoming z-stream.

File: microshift_release_notes/microshift-4-14-release-notes.adoc (hunk 2 above)
Comment (ShaunaDiaz) [reply microshift_release_notes/microshift-4-14-release-notes.adoc:495]: Yes, they will resolve once customers get the email with the Errata info.

**Pull request merged**